pkg/build/builder/dockerutil.go

```
@@ -39,15 +47,31 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 		opts.OutputStream = os.Stderr
 	}
 	var err error
-	for retries := 0; retries <= DefaultPushRetryCount; retries++ {
-		err = client.PushImage(opts, authConfig)
+
+	for retries := 0; retries <= MaxPushRetryCount; retries++ {
+
+		err := client.PushImage(opts, authConfig)
 		if err == nil {
 			return nil
 		}
-		if retries == DefaultPushRetryCount {
+
+		errMessage := fmt.Sprintf("%s", err)
+		var errInfo string
+
+		switch true {
+		case strings.Contains(errMessage, "ping attempt failed with error"):
+			if retries > DefaultPushRetryCount {
+				return err
+			}
+			errInfo = " cause Docker registry is unavailable,"
+		case strings.Contains(errMessage, "is already in progress"):
```

I would not hide the original error message